packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -355,6 +365,7 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       port: props.port ?? clusterEngineBindConfig.port,
       dbClusterParameterGroupName: clusterParameterGroupConfig?.parameterGroupName,
       associatedRoles: clusterAssociatedRoles.length > 0 ? clusterAssociatedRoles : undefined,
+      enableIamDatabaseAuthentication: Lazy.any({ produce: () => this.enableIamAuthentication }),
```

Comment:
Why use a Lazy here? Why not just `enableIamDatabaseAuthentication: props. enableIamAuthentication`?